spec/requests/registrations/user_profile_spec.rb

```diff
@@ -23,39 +23,104 @@
   end
 
   describe "POST /registrations/user-profile" do
-    it "creates an induction coordinator profile" do
-      post "/registrations/user-profile", params: { user: {
-        full_name: full_name,
-        email: email,
-      } }
-
-      expect(created_user.full_name).to eq(full_name)
-      expect(response).to redirect_to(:registrations_verification_sent)
-      expect(created_user.induction_coordinator_profile.schools).to include(school)
+    before do
+      mail = instance_double(ActionMailer::MessageDelivery, deliver_now: true)
+      allow(UserMailer).to receive(:confirmation_instructions).and_return(mail)
+    end
+
+    context "when request is valid" do
+      let(:send_request) do
+        post "/registrations/user-profile", params: { user: {
+          full_name: full_name,
+          email: email,
+        } }
+      end
```

Comment:
What does everyone think about this being a `let`? I personally would make this a private method, but I'd like to know what the consensus is. Since this is per-test memoised, it feels wrong to use `let` for side effects

Reply:
Looking at the implementation, this maybe lives in a `before` block? (I think I would still put it in a private method too in order to give it a nice name)

Reply:
I've seen let used for this sort of thing loads of times too but I think a private method would be a better idea as it encapsulates an action / behaviour

Reply:
It doesn't really live in a `before` block, sadly.

```
context "when school has old unconfirmed registrations" do
```
context needs to do its `before` block before sending off a request. 

I guess I could separate it out from the `when request is valid` context though...